artiq/dashboard/datasets.py

```diff
@@ -20,32 +21,180 @@ async def rename(key, new_key, value, persist, dataset_ctl):
     await dataset_ctl.set(new_key, value, persist)
 
 
+class AutoEditor:
+    def __init__(self):
+        self.widget = QtWidgets.QLineEdit()
+        self.widget.setPlaceholderText('PYON (Python)')
+
+    def get_value(self):
+        return pyon.decode(self.widget.text())
+
+    def set(self, val):
+        self.widget.setText(pyon.encode(val))
+
+    def connect(self, f):
+        self.widget.textChanged.connect(f)
+
+
+class NoneEditor:
+    def __init__(self):
+        self.widget = QtWidgets.QLabel("None")
+
+    def get_value(self):
+        return None
+
+    def connect(self, f):
+        pass
+
+    def set(self, val):
+        pass
+
+
+class BoolEditor:
+    def __init__(self):
+        self.widget = QtWidgets.QCheckBox()
+        self.widget.setChecked(False)
+
+    def get_value(self):
+        return self.widget.isChecked()
+
+    def set(self, val):
+        if type(val) is str and val.lower() == "false":
```

Comment:
If user inputs a string value, and then changes the type to bool, this case would transfer `"false"` into `False`

Reply:
Since any non-empty string is True, only `"false"` needs to be excepted in this case. 
Moreover, I think end user would be bit surprised after entering `"false"` into a string field, converting it to the bool, and seeing True.

Reply:
Ok, since this doesn't even happen in JS, removed this feature